src/System.Private.ServiceModel/src/System/ServiceModel/NetHttpsBinding.cs

```diff
@@ -0,0 +1,147 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.ServiceModel.Channels;
+using System.Xml;
+
+namespace System.ServiceModel
+{
+    public class NetHttpsBinding : HttpBindingBase
+    {
+        BinaryMessageEncodingBindingElement _binaryMessageEncodingBindingElement;
+        NetHttpMessageEncoding _messageEncoding;
+        BasicHttpsSecurity _basicHttpsSecurity;
+
+        public NetHttpsBinding() : this(BasicHttpsSecurity.DefaultMode) { }
+
+        public NetHttpsBinding(BasicHttpsSecurityMode securityMode)
+        {
+            if (securityMode == BasicHttpsSecurityMode.TransportWithMessageCredential)
+            {
+                throw ExceptionHelper.PlatformNotSupported(SR.Format(SR.UnsupportedSecuritySetting, "securityMode", securityMode));
+            }
+
+            Initialize();
+            _basicHttpsSecurity.Mode = securityMode;
+        }
+
+        [ComponentModel.DefaultValue(NetHttpMessageEncoding.Binary)]
+        public NetHttpMessageEncoding MessageEncoding
+        {
+            get { return _messageEncoding; }
+            set { _messageEncoding = value; }
+        }
+
+        public BasicHttpsSecurity Security
+        {
+            get
+            {
+                return _basicHttpsSecurity;
+            }
+
+            set
+            {
+                if (value == null)
+                {
+                    throw FxTrace.Exception.ArgumentNull("value");
+                }
+
+                _basicHttpsSecurity = value;
```

Comment:
This will be caught as soon as a channel factory is attempted to be made as CheckSettings in HttpBindingBase will get called as part of that. When I write the unit tests, I'll make sure to verify TransportWithMessageCredential fails.